protocol/app/ante/market_update_test.go

```diff
@@ -425,18 +443,117 @@ func TestValidateMarketUpdateDecorator_AnteHandle(t *testing.T) {
 			},
 			wantErr: true,
 		},
+		{
+			name: "reject a single message with a new market that's enabled",
+			args: args{
+				msgs: []sdk.Msg{
+					&mmtypes.MsgUpsertMarkets{
+						Authority: constants.BobAccAddress.String(),
+						Markets: []mmtypes.Market{
+							testMarketWithEnabled,
+						},
+					},
+				},
+				simulate:         false,
+				marketMapMarkets: []mmtypes.Market{testMarket}, // existing mm market is disabled
+			},
+			wantErr: true,
+		},
+		{
+			name: "reject a single message with a new market that's enabled - simulate",
+			args: args{
+				msgs: []sdk.Msg{
+					&mmtypes.MsgUpsertMarkets{
+						Authority: constants.BobAccAddress.String(),
+						Markets: []mmtypes.Market{
+							testMarketWithEnabled,
+						},
+					},
+				},
+				simulate: true,
+			},
+			wantErr: true,
+		},
+		{
+			name: "reject single message disabling a market",
+			args: args{
+				msgs: []sdk.Msg{
+					&mmtypes.MsgUpsertMarkets{
+						Authority: constants.BobAccAddress.String(),
+						Markets: []mmtypes.Market{
+							testMarket,
+						},
+					},
+				},
+				simulate:         false,
+				marketMapMarkets: []mmtypes.Market{testMarketWithEnabled},
+				marketParams:     []prices_types.MarketParam{testMarketParams},
+			},
+			wantErr: true,
+		},
+		{
+			name: "reject single message disabling a market - simulate",
+			args: args{
+				msgs: []sdk.Msg{
+					&mmtypes.MsgUpsertMarkets{
+						Authority: constants.BobAccAddress.String(),
+						Markets: []mmtypes.Market{
+							testMarket,
+						},
+					},
+				},
+				simulate:         true,
+				marketMapMarkets: []mmtypes.Market{testMarketWithEnabled},
+				marketParams:     []prices_types.MarketParam{testMarketParams},
+			},
+			wantErr: true,
+		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			tApp := testapp.NewTestAppBuilder(t).Build()
 			ctx := tApp.InitChain()
 
+			for _, mp := range tt.args.marketParams {
+				marketID := rand.Uint32()
+				mp.Id = marketID
+				_, err := tApp.App.PricesKeeper.CreateMarket(ctx, mp, prices_types.MarketPrice{
+					Id:       marketID,
+					Exponent: -8,
+					Price:    10,
+				})
+				require.NoError(t, err)
+			}
+
+			// setup initial market-map markets
+			for _, market := range tt.args.marketMapMarkets {
```

Comment:
Can you move this to before creating the market in x/prices? I'm adding a dependency when creating the market in x/prices that the market needs to exist in market map